platform/darwin/src/MGLStyle.mm

```diff
@@ -282,18 +282,19 @@ - (void)insertObject:(MGLStyleLayer *)styleLayer inLayersAtIndex:(NSUInteger)ind
          styleLayer];
     }
     auto layers = self.mapView.mbglMap->getLayers();
-    if (index > layers.size()) {
+    if (index >= layers.size()) {
         [NSException raise:NSRangeException
                     format:@"Cannot insert style layer at out-of-bounds index %lu.", (unsigned long)index];
     } else if (index == 0) {
```

Comment:
`index == layers.size()` gets caught by the case above, which I think is incorrect: you can append a single object to NSMutableArray by passing its `count` into `-insertObject:atIndex:`; an exception only gets raised if the index is `count + 1` or higher.